### PR TITLE
Features/select all

### PR DESCRIPTION
**Pull request by ZiQueue**
- Select all working for rows in Input Systems

### CODE REVIEW

File: src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.ts [hunk 1]
```diff
@@ -32,6 +32,27 @@ export class UnifiedConfigurationController implements angular.IController {
       this.unifiedViewModel = new ConfigurationUnifiedViewModel(this.uccConfigDirty);
     }
   }
+
+  selectAll(inputSystems: any): void {
+    const index = this.unifiedViewModel.inputSystems.indexOf(inputSystems);
+    const roles = RoleType.roles();
+    for (const role of roles) {
```
Comment (irahopkinson) [inline src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.ts]: You are iterating over all the roles but not over any of the groups. Need to add a _for_ loop for groups.

File: src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.ts [hunk 2]
```diff
@@ -32,6 +32,27 @@ export class UnifiedConfigurationController implements angular.IController {
...
+        this.unifiedViewModel.inputSystems[index][role] = this.unifiedViewModel.inputSystems[index].ifAllRowSelected;
+    }
+  }
+
+  checkIfAllSelected(inputSystems: any): void {
+    const index = this.unifiedViewModel.inputSystems.indexOf(inputSystems);
+    const roles = RoleType.roles();
+    for (const role of roles) {
```
Comment (irahopkinson) [inline src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.ts]: You are iterating over all the roles but not over any of the groups. Need to add a _for_ loop for groups.

File: src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified-view.model.ts [hunk 3]
```diff
@@ -145,13 +145,14 @@ export class InputSystemSettings {
   contributor: boolean;
   manager: boolean;
   groups: boolean[] = [];
+  ifAllRowSelected: boolean;
```
[the remark this answers is not in the document]
Comment (ZiQueue) [reply src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified-view.model.ts]: done

File: src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.html [hunk 4]
```diff
@@ -30,14 +30,14 @@
                 </tr>
                 <tr data-ng-repeat="inputSystems in $ctrl.unifiedViewModel.inputSystems">
                     <th class="text-center table-secondary" ></th>
-                    <td class="text-center"><input class="form-check-input position-static" data-ng-model="input.hiddenIfEmpty" aria-label="..." type="checkbox"></td>
+                    <td class="text-center"><input class="form-check-input position-static" data-ng-model="inputSystems.hiddenIfEmpty" aria-label="..." type="checkbox"></td>
```
[the remark this answers is not in the document]
Comment (ZiQueue) [reply src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.html]: done

File: src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.ts [hunk 5]
```diff
@@ -32,6 +32,27 @@ export class UnifiedConfigurationController implements angular.IController {
       this.unifiedViewModel = new ConfigurationUnifiedViewModel(this.uccConfigDirty);
     }
   }
+
+  selectAll(inputSystems: any): void {
```
[the remark this answers is not in the document]
Comment (ZiQueue) [reply src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.ts]: done

File: src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.ts (hunk 1 above)
Comment (ZiQueue) [reply src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.ts]: done

File: src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.ts [hunk 6]
```diff
@@ -32,6 +32,27 @@ export class UnifiedConfigurationController implements angular.IController {
...
+    const index = this.unifiedViewModel.inputSystems.indexOf(inputSystems);
+    const roles = RoleType.roles();
+    for (const role of roles) {
+        this.unifiedViewModel.inputSystems[index][role] = this.unifiedViewModel.inputSystems[index].ifAllRowSelected;
+    }
+  }
+
+  checkIfAllSelected(inputSystems: any): void {
```
[the remark this answers is not in the document]
Comment (ZiQueue) [reply src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.ts]: done

File: src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.ts (hunk 2 above)
Comment (ZiQueue) [reply src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.ts]: done

File: src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.ts [hunk 7]
```diff
@@ -32,6 +32,27 @@ export class UnifiedConfigurationController implements angular.IController {
...
+    const index = this.unifiedViewModel.inputSystems.indexOf(inputSystems);
+    const roles = RoleType.roles();
+    for (const role of roles) {
+      if (this.unifiedViewModel.inputSystems[index][role] == false) {
+          this.unifiedViewModel.inputSystems[index].ifAllRowSelected = false;
+          break;
+      }
+      this.unifiedViewModel.inputSystems[index].ifAllRowSelected = true;
+    }
+    console.log(this.unifiedViewModel.inputSystems[index].ifAllRowSelected);
```
[the remark this answers is not in the document]
Comment (ZiQueue) [reply src/angular-app/languageforge/lexicon/settings/configuration/configuration-unified.component.ts]: done

**Pull request merged**